modules/monitoring/models/scheduledate.php

```diff
@@ -1,479 +1,485 @@
+
+				// Submit the command.
+				// Note: submit_to_nagios has its own internal /error/ handling.
+				$result = nagioscmd::submit_to_nagios($command);
+
+				if($result) { // success
+					$start_dt = $downtime->start->get_datetime();
+					$end_dt = $downtime->end->get_datetime();
+					$this->logger->log('debug', "[$comment] Downtime scheduled for object: $name ($start_dt - $end_dt)");
+				}
+
+				array_push($results, $result);
 			}
 		}
-	}
-	return !in_array(false, $result);
-}
 
-/**
-* Schedule a downtime by submitting it to nagios
-*
-* @param $objects array
-* @param $object_type string
-* @param $start_time string
-* @param $end_time string
-* @param $start_date string
-* @param $end_date string
-* @param $fixed string
-* @param $duration string
-* @param $comment string
-* @return boolean
-**/
-public static function insert_downtimes($objects, $object_type, $start_time, $end_time, $start_date, $end_date, $fixed, $duration, $comment)
-{
-	$result = array();
-	$nagios_cmd = self::determine_downtimetype($object_type);
-	$author = op5auth::instance()->get_user()->get_username();
-	$month = date('n');
-	$day = date('d');
-	$year = date('Y');
-	$strt_d = explode('-',$start_date);
-	$sy = (int)$strt_d[0];
-	$sm = (int)$strt_d[1];
-	$sd = (int)$strt_d[2];
-	$end_d = explode('-',$end_date);
-	$ey = (int)$end_d[0];
-	$em = (int)$end_d[1];
-	$ed = (int)$end_d[2];
-	$start_time = mktime(0, 0, self::time_to_seconds($start_time), $sm, $sd, $sy);
-	$end_time = mktime(0, 0, self::time_to_seconds($end_time), $em, $ed, $ey);
-	foreach ($objects as $object) {
-		if (static::check_if_scheduled($object_type, $object, $start_time, $end_time, $fixed)) {
-		// Skip object if it is already scheduled for downtime
-			continue;
-		}
-		$tmp_cmd = "$nagios_cmd;$object;$start_time;$end_time;$fixed;0;$duration;$author;AUTO: $comment";
-		$result[] = nagioscmd::submit_to_nagios($tmp_cmd);
+		// Return false if one or more external commands failed, otherwise true.
+		return !in_array(false, $results);
 	}
-	return !in_array(false, $result);
-}
 
-/**
-* Given a time-like string (hh[:mm[:ss]]),
-* return the number of seconds involved.
-*/
-static public function time_to_seconds($time)
-{
-	$seconds = 0;
-	$parts = explode(':', $time);
-	if (isset($parts[0]))
-		$seconds += $parts[0] * 3600;
-	if (isset($parts[1]))
-		$seconds += $parts[1] * 60;
-	if (isset($parts[2]))
-		$seconds += $parts[2];
-	return $seconds;
-}
+	/**
+	 * Use a reasonable amount of indicators to determine whether there's
+	 * already a matching downtime. This prevents downtimes from being
+	 * scheduled more than once.
+	 */
+	static protected function check_if_scheduled($type, $name, $start_time, $end_time, $is_fixed)
+	{
+		$ls = Livestatus::instance();
+		switch ($type) {
+			case 'hosts':
+				$res = $ls->getDowntimes(array('filter' => array('is_service' => 0, 'host_name' => $name, 'start_time' => $start_time, 'fixed' => $is_fixed, 'end_time' => $end_time)));
```

Comment:
Not sure how long this line are. But if possible, I would like to break it down to more and shorter rows, to make it more readable. According to our coding standard [line lenght](https://kb.op5.com/display/DEV/MON+-+PHP+Coding+Standard) should be around 80 columns.
Same comment for cases below

Reply:
That's not from this commit, it's probably been there for ages. I haven't bothered updating these parts - but I do agree that this should be addressed.